test/framework-ondata-mocha.js

```diff
@@ -1048,102 +1063,159 @@ should.be.rejectedWith('No wrapper fRunUpdater function found!  Cannot run updat
                                                  }}
                 )},
-                downstreamNode: 'rdf-components/vni-data-output' 
+                downstreamNode: {getComponent: factory({inPorts: {input: {}},
+                                                        outPorts: {output: {}},
+                                                        isTransient: true
+                                                       },
+                                                       {fRunUpdater: function(vni, args) { 
+
+                                                         nodeInstance = vni.nodeInstance;
+                                                         nodeInstance.vnis.should.be.an('object');
+                                                         nodeInstance.vnis.should.have.all.keys('1'); 
+
+                                                         var inputStates = vni.inputStates();
+                                                         inputStates.should.have.all.keys('input');
+
+                                                         test.verifyState(inputStates.input, '1', initData);
+
+                                                         vni.outputState({vnid:'1',
+                                                                          data: inputStates.input.data + 
+                                                                                ' ' + downstreamData,
+                                                                          lm: createLm()});    
+                                                 }}
+                )}
             }).then(function(network){
 
                  var testNode = network.processes.testNode.component;
                  var downstreamNode = network.processes.downstreamNode.component;
 
-                 return new Promise(function(done, fail) {
+                 return new Promise(function(done) {
 
-                     network.graph.addEdge('testNode', 'output', 'downstreamNode', 'in');
+                     network.graph.addEdge('testNode', 'output', 'downstreamNode', 'input');
 
-		     test.onOutPortData(downstreamNode, 'out', done);
+		     test.onOutPortData(downstreamNode, 'output', done);
 
-                     sinon.stub(console,'log');
-                     network.graph.addInitial("init data", 'testNode', 'input');
+                     network.graph.addInitial(initData, 'testNode', 'input');
 
                  }).then(function(done) {
-                     console.log.restore();
-                     done.should.equal(testData);
+                     test.verifyState(done, '1', 'Buon pomeriggio! Arrivederci!');
 
                      // Verify VNI has been cleared
                      expect(nodeInstance).to.not.be.empty;
                      nodeInstance.vnis.should.be.an('object');
-                     nodeInstance.vnis.should.be.empty;
+
+                     var states = nodeInstance.vni('1').inputStates(); 
```

Comment:
There's not a real easy way to do that directly since as soon as you do nodeInstance.vni('1') it will create it.  However, I did rework this based on this comment.  I deleted the inputState check and replaced with:

                     Object.keys(nodeInstance.vnis).should.be.empty;

Since in this case the VNI has been deleted, there should be no VNI keys.